### PR TITLE
Update

### PR DESCRIPTION
**Pull request by Artsav23**

### CODE REVIEW

**Comment by mokhaneugene** [comment]
What is "Update" at the title of pr (pull request)?
Write the name of the changes at this pr pls

File: app/src/main/java/com/example/socialnetwork/Adapter.kt [hunk 1]
```diff
@@ -6,43 +6,46 @@ import android.view.ViewGroup
 import androidx.recyclerview.widget.RecyclerView
 import com.example.socialnetwork.databinding.ViewBinding
 
-class Adapter(val listener: Listener): RecyclerView.Adapter<Adapter.ViewHolder>() {
-    var dataView= mutableListOf<DataView>()
+class Adapter(private val listener: Listener): RecyclerView.Adapter<Adapter.ViewHolder>() {
+    var publicationModel= mutableListOf<PublicationModel>()
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/Adapter.kt:10]: I think, you should set private at the 10 line, not at the 9 line, but mb(maybe) it's incorrect

File: app/src/main/java/com/example/socialnetwork/Adapter.kt [hunk 2]
```diff
@@ -6,43 +6,46 @@ import android.view.ViewGroup
...
 
     class ViewHolder(item:View):RecyclerView.ViewHolder(item) {
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/Adapter.kt:12]: Why did you create the class inside other class?

File: app/src/main/java/com/example/socialnetwork/Adapter.kt [hunk 3]
```diff
@@ -6,43 +6,46 @@ import android.view.ViewGroup
...
         var num = 1
         var binding=ViewBinding.bind(item)
-        fun bind(dataView: DataView, listener: Listener){
-            binding.imageView3.setImageResource(dataView.imageId)
-            binding.comments.text=dataView.comment
+
+        fun bind(publicationModel: PublicationModel, listener: Listener){
+            binding.imageView3.setImageResource(publicationModel.imageId)
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/Adapter.kt:17]: what is imageView3? Change the name of this view

File: app/src/main/java/com/example/socialnetwork/Adapter.kt [hunk 4]
```diff
@@ -6,43 +6,46 @@ import android.view.ViewGroup
...
+            binding.comments.text=publicationModel.comment
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/Adapter.kt:18]: try to use the type of the view at the naming, for example, when you create a textView for the title, try to use titleTextView

File: app/src/main/java/com/example/socialnetwork/Adapter.kt [hunk 5]
```diff
@@ -6,43 +6,46 @@ import android.view.ViewGroup
...
     }
-    fun add (data: DataView){
-        dataView.add(data)
+
+    fun add (data: PublicationModel){
+        publicationModel.add(data)
         notifyDataSetChanged()
     }
+
     interface Listener{
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/Adapter.kt:48]: what is interface?

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 6]
```diff
@@ -5,37 +5,40 @@ import androidx.appcompat.app.AppCompatActivity
...
 import android.view.Menu
 import android.view.MenuItem
-import androidx.activity.result.contract.ActivityResultContracts
 import com.example.socialnetwork.databinding.ActivityAddPudlicationBinding
-import com.example.socialnetwork.databinding.ActivityMainBinding
 
-class AddPudlication : AppCompatActivity() {
+
+class AddPublication : AppCompatActivity() {
     lateinit var binding: ActivityAddPudlicationBinding
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/AddPublication.kt:12]: set the empty line pls

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 7]
```diff
@@ -5,37 +5,40 @@ import androidx.appcompat.app.AppCompatActivity
...
     var images=Images()
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/AddPublication.kt:13]: why it isn't private?

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 8]
```diff
@@ -5,37 +5,40 @@ import androidx.appcompat.app.AppCompatActivity
...
+        setSupportActionBar()
+
         binding.next.setOnClickListener {
             if (num==4){
                 num=0
             }
-            else{
+
+            else {
             num++
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/AddPublication.kt:28]: fix the tabulation pls

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 9]
```diff
@@ -5,37 +5,40 @@ import androidx.appcompat.app.AppCompatActivity
...
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding=ActivityAddPudlicationBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        binding.imageView.setImageResource(images.images[num])
-        supportActionBar?.title="Add Publication"
-        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        setSupportActionBar()
+
         binding.next.setOnClickListener {
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/AddPublication.kt:22]: can u (you) set this logic at other function?

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 10]
```diff
@@ -5,37 +5,40 @@ import androidx.appcompat.app.AppCompatActivity
...
             binding.imageView.setImageResource(images.images[num])
         }
+
         binding.cancel.setOnClickListener {
             if (num==0){
                 num=4
             }
+
             else{
             num--
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/AddPublication.kt:39]: fix the tabulation pls

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 11]
```diff
@@ -5,37 +5,40 @@ import androidx.appcompat.app.AppCompatActivity
...
             }
-            else{
+
+            else {
             num++
             }
             binding.imageView.setImageResource(images.images[num])
         }
+
         binding.cancel.setOnClickListener {
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/AddPublication.kt:33]: can u set this logic at other function?

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 12]
```diff
@@ -5,37 +5,40 @@ import androidx.appcompat.app.AppCompatActivity
...
             if (num==0){
                 num=4
             }
+
             else{
             num--
             }
+
             binding.imageView.setImageResource(images.images[num])
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/AddPublication.kt:42]: can u set this logic at other function?

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 13]
```diff
@@ -47,8 +50,9 @@ class AddPudlication : AppCompatActivity() {
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         when (item.itemId){
+
             R.id.add->{
-                var data =Intent()
+                val data =Intent()
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/AddPublication.kt:55]: set the space between = and Intent()

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 14]
```diff
@@ -57,4 +61,9 @@ class AddPudlication : AppCompatActivity() {
         }
         return true
     }
+
+    private fun setSupportActionBar(){
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/AddPublication.kt:65]: set the space between () and {

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 15]
```diff
@@ -57,4 +61,9 @@ class AddPudlication : AppCompatActivity() {
...
+        supportActionBar?.title="Add Publication"
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/AddPublication.kt:66]: supportActionBar?.title = "Add Publication"

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt [hunk 16]
```diff
@@ -4,28 +4,25 @@ import android.content.Intent
...
 import android.view.Menu
-import android.view.View
-import android.widget.Toast
 import androidx.activity.result.ActivityResultLauncher
 import androidx.activity.result.contract.ActivityResultContracts
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.example.socialnetwork.databinding.ActivityMainBinding
 
 class MainActivity : AppCompatActivity(), Adapter.Listener {
     lateinit var launcher: ActivityResultLauncher<Intent>
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/MainActivity.kt:13]: can it been private?

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt [hunk 17]
```diff
@@ -4,28 +4,25 @@ import android.content.Intent
...
-    lateinit var launcherView: ActivityResultLauncher<Intent>
     lateinit var binding: ActivityMainBinding
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/MainActivity.kt:14]: can it been private?

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt [hunk 18]
```diff
@@ -4,28 +4,25 @@ import android.content.Intent
...
     var adapter=Adapter(this)
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/MainActivity.kt:15]: can it been private?

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt [hunk 19]
```diff
@@ -4,28 +4,25 @@ import android.content.Intent
...
-    var data= mutableListOf<DataView>()
+    var data= mutableListOf<PublicationModel>()
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/MainActivity.kt:16]: can it been private?

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt [hunk 20]
```diff
@@ -4,28 +4,25 @@ import android.content.Intent
...
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        supportActionBar?.title="Messeger"
 
         launcher=registerForActivityResult(ActivityResultContracts.StartActivityForResult()){
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/MainActivity.kt:23]: can it locate at other function?

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt [hunk 21]
```diff
@@ -34,32 +31,34 @@ class MainActivity : AppCompatActivity(), Adapter.Listener {
...
+        initRecyclerView()
     }
 
     override fun onCreateOptionsMenu(menu: Menu?): Boolean {
         menuInflater.inflate(R.menu.main_menu, menu)
         return true
     }
-    fun init(){
+
+    private fun initRecyclerView(){
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/MainActivity.kt:50]: private fun initRecyclerView() {

**Review by mokhaneugene — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/socialnetwork/Adapter.kt (hunk 5 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/Adapter.kt:48]: хрень позволяющая делать действия когда твое вью открыто на другом вью, в данном случае она позволяет мне при нажатии на публикацию перейти на отдельную страницу с одним вью

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt (hunk 9 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/AddPublication.kt:22]: не понял

File: app/src/main/java/com/example/socialnetwork/Adapter.kt (hunk 1 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/Adapter.kt]: сделал

File: app/src/main/java/com/example/socialnetwork/Adapter.kt (hunk 3 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/Adapter.kt]: сделал

File: app/src/main/java/com/example/socialnetwork/Adapter.kt (hunk 4 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/Adapter.kt]: сделал

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt (hunk 6 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/AddPublication.kt]: сделал

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt (hunk 7 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/AddPublication.kt]: сделал

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt (hunk 8 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/AddPublication.kt]: сделал

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt (hunk 10 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/AddPublication.kt]: сделал

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt (hunk 12 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/AddPublication.kt]: сделал

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt [hunk 22]
```diff
@@ -4,28 +4,25 @@ import android.content.Intent
...
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        supportActionBar?.title="Messeger"
 
         launcher=registerForActivityResult(ActivityResultContracts.StartActivityForResult()){
                 if (it.resultCode== RESULT_OK){
-                    data.add(DataView(it.data?.getStringExtra("imageId")?.toInt()!!,
+                    data.add(PublicationModel(it.data?.getStringExtra("imageId")?.toInt()!!,
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/MainActivity.kt:26]: работать не будет тогда

File: app/src/main/java/com/example/socialnetwork/OneView.kt [hunk 23]
```diff
@@ -9,16 +9,15 @@ import com.example.socialnetwork.databinding.ActivityOneViewBinding
 
 class OneView : AppCompatActivity() {
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/OneView.kt:10]: окно которое показывает публикацию в отдельном окне

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt (hunk 21 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/MainActivity.kt]: сделал

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt [hunk 24]
```diff
@@ -34,32 +31,34 @@ class MainActivity : AppCompatActivity(), Adapter.Listener {
         binding.bNavView.setOnItemSelectedListener {
             when(it.itemId){
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/MainActivity.kt]: сделал

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt [hunk 25]
```diff
@@ -34,32 +31,34 @@ class MainActivity : AppCompatActivity(), Adapter.Listener {
...
                R.id.add->{
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/MainActivity.kt]: сделал

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt [hunk 26]
```diff
@@ -4,28 +4,25 @@ import android.content.Intent
...
+    var data= mutableListOf<PublicationModel>()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        supportActionBar?.title="Messeger"
 
         launcher=registerForActivityResult(ActivityResultContracts.StartActivityForResult()){
                 if (it.resultCode== RESULT_OK){
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/MainActivity.kt]: done

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt (hunk 19 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/MainActivity.kt]: done

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt (hunk 18 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/MainActivity.kt]: done

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt (hunk 16 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/MainActivity.kt]: done

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt (hunk 14 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/AddPublication.kt]: done

**Review by mokhaneugene — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt [hunk 27]
```diff
@@ -4,62 +4,62 @@ import android.content.Intent
...
                 }
         }
 
         binding.bNavView.setOnItemSelectedListener {
-            when(it.itemId){
-               R.id.add->{
-                   var intent= Intent(this, AddPudlication::class.java)
+            when(it.itemId) {
+               R.id.add-> {
+                   val intent= Intent(this, AddPublication::class.java)
```
Comment (mokhaneugene) [inline app/src/main/java/com/example/socialnetwork/MainActivity.kt:35]: val intent =

**Review by mokhaneugene — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt (hunk 14 above)
Comment (mokhaneugene) [reply app/src/main/java/com/example/socialnetwork/AddPublication.kt]: пробел поставь

File: app/src/main/java/com/example/socialnetwork/OneView.kt [hunk 28]
```diff
@@ -9,16 +9,15 @@ import com.example.socialnetwork.databinding.ActivityOneViewBinding
...
     lateinit var binding: ActivityOneViewBinding
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/OneView.kt]: done

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt (hunk 27 above)
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/MainActivity.kt]: done

File: app/src/main/java/com/example/socialnetwork/ViewHolder.kt [hunk 29]
```diff
@@ -0,0 +1,24 @@
...
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import com.example.socialnetwork.databinding.ViewBinding
+
+class ViewHolder(item: View): RecyclerView.ViewHolder(item) {
+
+    var num = 1
+    var binding= ViewBinding.bind(item)
+
+    fun bind(publicationModel: PublicationModel, listener: Adapter.Listener){
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/ViewHolder.kt]: done

File: app/src/main/java/com/example/socialnetwork/PublicationModel.kt [hunk 30]
```diff
@@ -0,0 +1,6 @@
+package com.example.socialnetwork
+
+import java.io.Serializable
+
+
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/PublicationModel.kt]: done

File: app/src/main/java/com/example/socialnetwork/OneView.kt [hunk 31]
```diff
@@ -8,22 +8,22 @@ import android.widget.Toast
...
-        Toast.makeText(this,"!!!", Toast.LENGTH_SHORT).show()
-        val item=intent.getSerializableExtra("item") as DataView
-        binding.imageView3.setImageResource(item.imageId)
-        binding.comments.text=item.comment
-//        binding.imageView3.setImageResource(intent.getStringExtra("imageId")?.toInt()!!)
-//        binding.comments.text=intent.getStringExtra("comment")
+
+        val item=intent.getSerializableExtra("item") as PublicationModel
+        binding.image.setImageResource(item.imageId)
+        binding.commentsTextView.text=item.comment
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/OneView.kt]: done

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 32]
```diff
@@ -57,4 +47,30 @@ class AddPudlication : AppCompatActivity() {
...
+            num = 4
+        }
+        else{
+            num--
+        }
+
+        binding.imageView.setImageResource(images.images[num])
+    }
+
+    fun next() {
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/AddPublication.kt]: done

File: app/src/main/java/com/example/socialnetwork/AddPublication.kt [hunk 33]
```diff
@@ -57,4 +47,30 @@ class AddPudlication : AppCompatActivity() {
         }
         return true
     }
+
+    private fun setSupportActionBar() {
+        supportActionBar?.title = "Add Publication"
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    }
+
+    fun cancel() {
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/AddPublication.kt]: done

File: app/src/main/java/com/example/socialnetwork/OneView.kt [hunk 34]
```diff
@@ -9,16 +9,15 @@ import com.example.socialnetwork.databinding.ActivityOneViewBinding
...
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding= ActivityOneViewBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        Toast.makeText(this,"!!!", Toast.LENGTH_SHORT).show()
-        val item=intent.getSerializableExtra("item") as DataView
+
+        val item=intent.getSerializableExtra("item") as PublicationModel
         binding.imageView3.setImageResource(item.imageId)
         binding.comments.text=item.comment
```
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/OneView.kt]: done

File: app/src/main/java/com/example/socialnetwork/OneView.kt (hunk 23 above)
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/OneView.kt]: done

File: app/src/main/java/com/example/socialnetwork/MainActivity.kt (hunk 22 above)
[the remark this answers is not in the document]
Comment (Artsav23) [reply app/src/main/java/com/example/socialnetwork/MainActivity.kt]: done

**Review by mokhaneugene — APPROVED** [review]
Good work!

**Pull request merged**